tests/cephfs/cephfs_scale/max_subvolumes.py

```diff
@@ -0,0 +1,113 @@
+    )
+
+
+def collect_ceph_details(client, cmd):
+    out, rc = client.exec_command(sudo=True, cmd=f"{cmd} --format json-pretty")
+    output = json.loads(out)
+    log.info(output)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test Cases Covered:
+    CEPH-83573520	Validate the max snapshot that can be created under a root FS sub volume level.
+
+    Pre-requisites :
+    1. We need atleast one client node to execute this test case
+    2. creats fs volume create cephfs if the volume is not there
+    3. ceph fs subvolume create <vol_name> <subvol_name> [--size <size_in_bytes>] [--group_name <subvol_group_name>]
+       [--pool_layout <data_pool_name>] [--uid <uid>] [--gid <gid>] [--mode <octal_mode>]  [--namespace-isolated]
+       Ex: ceph fs subvolume create cephfs subvol_max_snap --size 5368706371 --group_name subvolgroup_1
+    4. Create Data on the subvolume
+        Ex:  python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 400 --files
+            100 --files-per-dir 10 --dirs-per-dir 2 --top /mnt/cephfs_fuse1baxgbpaia_1/
+
+    Test Script Flow :
+    1. We will get the total available space.
+    2. Based on total Available space we create a number of volumes with 1 GB each
+    3. Verify there is no error after that
+
+    Clean up:
+    1. Deletes all the subvolumes created
+    """
+    try:
+        fs_util = FsUtils(ceph_cluster)
+        config = kw.get("config")
+        clients = ceph_cluster.get_ceph_objects("client")
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+        default_fs = "cephfs"
+        client1 = clients[0]
+        available_space = int(get_available_space(client1, default_fs))
+        log.info(available_space)
+        data_size_iter = int(available_space / 1024000000)
+        subvol_list = [
+            {
+                "vol_name": default_fs,
+                "subvol_name": f"subvol_max_{x}",
+            }
+            for x in range(0, data_size_iter)
+        ]
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        for idx, subvol in enumerate(subvol_list):
+            try:
+                fs_util.create_subvolume(clients[0], **subvol, validate=False)
+                fuse_mounting_dir_1 = f"/mnt/cephfs_fuse{mounting_dir}_{idx}/"
+                fs_util.fuse_mount([clients[0]], fuse_mounting_dir_1)
+                clients[0].exec_command(
+                    sudo=True,
+                    cmd=f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 "
+                    f"--file-size 1024 "
+                    f"--files 1024 --files-per-dir 10 --dirs-per-dir 1 --top "
+                    f"{fuse_mounting_dir_1}",
+                    long_running=True,
+                )
```

Comment:
Hey @AmarnatReddy , could you check if this small file is creating 1 GB data. In logs I'm seeing very less data in 'total_used_bytes' in "ceph df" & 'used'  in "ceph fs status"

Reply:
latest logs for this Yogesh. I made a few changes to that
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-3P0UB4/

Can you review it again